runtime/compiler/z/codegen/J9S390JNILinkage.cpp

```diff
@@ -0,0 +1,927 @@
+
+   TR::S390CHelperLinkage* linkage = static_cast<TR::S390CHelperLinkage*>(cg()->getLinkage(TR_CHelper));
+   linkage->buildIndirectDispatch(helperCallNode, &deps, NULL, isJNICallOutFrame);
+   }
+
+/**
+  * \details
+  *
+  *  The 5-slot callout frame is structures as the following:
+  *
+  *  \verbatim
+  *
+  *        |-----|
+  *        |     |      <-- constJNICallOutFrameSpecialTag() (For jni thunk, constJNICallOutFrameInvisibleTag())
+  *  16/32 |-----|
+  *        |     |      <-- savedPC. Empty.
+  *  12/24 |-----|
+  *        |     |      <-- return address for JNI call
+  *   8/16 |-----|
+  *        |     |      <-- constJNICallOutFrameFlags()
+  *   4/8   -----
+  *        |     |      <-- ramMethod for the native method
+  *         -----       <-- stack pointer
+  *
+  * \endverbatim
+  */
+void
+TR::J9S390JNILinkage::setupJNICallOutFrame(TR::Node * callNode,
+   TR::RealRegister * javaStackPointerRealRegister,
+   TR::Register * methodMetaDataVirtualRegister,
+   TR::LabelSymbol * returnFromJNICallLabel,
+   TR::S390JNICallDataSnippet *jniCallDataSnippet,
+                                             TR::Register* tempReg)
+   {
+   TR::CodeGenerator * codeGen = cg();
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(fe());
+   TR::ResolvedMethodSymbol * cs = callNode->getSymbol()->castToResolvedMethodSymbol();
+   TR_ResolvedMethod * resolvedMethod = cs->getResolvedMethod();
+   TR::Instruction * cursor = NULL;
+
+   int32_t stackAdjust = (-5 * (int32_t)sizeof(intptrj_t));
+
+   cursor = generateRXYInstruction(codeGen, TR::InstOpCode::LAY, callNode, tempReg,
+                                   generateS390MemoryReference(javaStackPointerRealRegister, stackAdjust, codeGen), cursor);
+
+   // set up Java Thread
+   intptrj_t constJNICallOutFrameType = fej9->constJNICallOutFrameType();
+   TR_ASSERT( constJNICallOutFrameType < MAX_IMMEDIATE_VAL, "OMR::Z::Linkage::setupJNICallOutFrame constJNICallOutFrameType is too big for MVHI");
+
+   TR_ASSERT((fej9->thisThreadGetJavaFrameFlagsOffset() == fej9->thisThreadGetJavaLiteralsOffset() + TR::Compiler->om.sizeofReferenceAddress()) &&
+           fej9->thisThreadGetJavaLiteralsOffset() == fej9->thisThreadGetJavaPCOffset() + TR::Compiler->om.sizeofReferenceAddress()
+           , "The vmthread field order should be pc,literals,jitStackFrameFlags\n");
+
+   jniCallDataSnippet->setPC(constJNICallOutFrameType);
+   jniCallDataSnippet->setLiterals(0);
+   jniCallDataSnippet->setJitStackFrameFlags(0);
+
+   generateSS1Instruction(cg(), TR::InstOpCode::MVC, callNode, 3*(TR::Compiler->om.sizeofReferenceAddress()) - 1,
+                          generateS390MemoryReference(methodMetaDataVirtualRegister, fej9->thisThreadGetJavaPCOffset(), codeGen),
+                          generateS390MemoryReference(jniCallDataSnippet->getBaseRegister(), jniCallDataSnippet->getPCOffset(), codeGen));
+
+   // store out jsp
+   generateRXInstruction(codeGen, TR::InstOpCode::getStoreOpCode(), callNode, tempReg,
```

Comment:
Can we rename `codeGen` to `cg` to stay consistent with the rest of the codebase?

Reply:
Changed to `cg()`